zk-token-sdk/src/zk_token_proof_instruction.rs

```diff
@@ -73,7 +73,11 @@ impl ProofInstruction {
     }
 
     pub fn decode_type(input: &[u8]) -> Option<Self> {
-        FromPrimitive::from_u8(input[0])
```

Comment:
Probably could get a little more fancy here with `input.get(0).map(FromPrimative::from_u8)`

Reply:
Yep, that is more elegant, indeed.